# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: 108d5fb4418e
# Full Hash: 108d5fb4418e78e4dca0a746863e5784c6d80371
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-23 09:52:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r d8dea951a032 -r 108d5fb4418e layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Wed Apr 22 15:48:20 2020 +0000
+++ b/layout/printing/nsPrintJob.cpp	Wed Apr 22 15:48:17 2020 +0000
@@ -3195,7 +3195,7 @@
     // dump the views of the sub documents
     int32_t i, n;
     BrowsingContext* bc = nsDocShell::Cast(aDocShell)->GetBrowsingContext();
-    for (auto& child : bc->GetChildren()) {
+    for (auto& child : bc->Children()) {
       if (auto childDS = child->GetDocShell()) {
         DumpViews(childAsShell, out);
       }